services/tables/src/main/java/com/linkedin/openhouse/tables/audit/TableAuditAspect.java

```diff
@@ -195,6 +195,46 @@ protected ApiResponse<Void> auditDeleteTable(
     return result;
   }
 
+  /** Install the Around advice for deleteTable() method in OpenHouseTablesApiHandler */
+  @Around(
+      "execution("
+          + "com.linkedin.openhouse.common.api.spec.ApiResponse<Void> "
+          + "com.linkedin.openhouse.tables.api.handler.TablesApiHandler.renameTable(..)) "
+          + "&& args(fromDatabaseId, fromTableId, toDatabaseId, toTableId, actingPrincipal)")
+  protected ApiResponse<Void> auditRenameTable(
+      ProceedingJoinPoint point,
+      String fromDatabaseId,
+      String fromTableId,
+      String toDatabaseId,
+      String toTableId,
+      String actingPrincipal)
+      throws Throwable {
+    ApiResponse<Void> result = null;
+    TableAuditEvent fromEvent =
+        TableAuditEvent.builder()
+            .eventTimestamp(Instant.now())
+            .databaseName(fromDatabaseId)
+            .tableName(fromTableId)
+            .operationType(OperationType.RENAME_FROM)
+            .build();
+    TableAuditEvent toEvent =
+        TableAuditEvent.builder()
+            .eventTimestamp(Instant.now())
+            .databaseName(toDatabaseId)
+            .tableName(toTableId)
+            .operationType(OperationType.RENAME_TO)
+            .build();
+    try {
+      result = (ApiResponse<Void>) point.proceed();
+      buildAndSendEvent(fromEvent, OperationStatus.SUCCESS, null);
+      buildAndSendEvent(toEvent, OperationStatus.SUCCESS, null);
```

Comment:
Shall we update the TableAuditEvent to include two new columns i.e. renamedTableId and renamedDbId so that whole rename operation is captured in a single event and capture the operation type as just `rename`. Otherwise, there would be disconnect on src and dest of rename operation.

Reply:
This is a good point, I'd like to add this in a separate PR though. I was thinking it'd be beneficial to generalize this. 

I was thinking of adding metadata related to each TableAuditEvent that is specific to its operation. This is [BigQuery's approach](https://cloud.google.com/bigquery/docs/reference/auditlogs/rest/Shared.Types/BigQueryAuditMetadata) which I think works for us well too. 

This allows us to have tableAuditMetadata specific to the operation itself e.g. RENAME will have renameFrom/renameTo information, DELETE can have information if it's hard delete or softDelete and it's TTL, UPDATEs can have more information about rows added/deleted in the audit event.

Reply:
As discussed offline, we need to use service audit event when rename url would present to join so that we can get info on one table renamed to which specific table. Ok, lets see how the current operation of `rename from` and `rename to` is beneficial for delete use case as well. If needed we can address in a separate PR.